fabric-mining-levels-v0/src/main/java/net/fabricmc/fabric/mixin/tools/MixinItemStack.java

```diff
@@ -46,7 +47,13 @@ public void getBlockBreakingSpeed(BlockState state, CallbackInfoReturnable<Float
 		if (this.getItem() instanceof MiningToolItemAccessor) {
 			TriState triState = ToolManager.handleIsEffectiveOn((ItemStack) (Object) this, state);
 			if (triState != TriState.DEFAULT) {
-				info.setReturnValue(triState.get() ? ((MiningToolItemAccessor) this.getItem()).getMiningSpeed() : 1.0F);
+				float miningSpeed;
+				if (this.getItem() instanceof DynamicTool) {
+					miningSpeed = ((DynamicTool) this.getItem()).getMiningSpeed((ItemStack)(Object) this);
+				} else {
+					miningSpeed = ((MiningToolItemAccessor) this.getItem()).getMiningSpeed();
```

Comment:
Then we have this... What is `MiningToolItemAccessor`?

Should the `DynamicTool` interface be split up? One for mining speed, and another for mining level so you can selectively implement each one independently?

(it's trivial to add a third interface as a union of the two, for convenience's sake)

Reply:
Yeah, it's just not worth it. If you don't want to impl one, you can always just have it pass through to the tool's hardcoded stats.